src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationTracker.cs

```diff
@@ -165,10 +164,13 @@ SymbolKind.NetModule or
                         }
 
                         var newState = CompilationTrackerState.Create(
-                            compilationWithoutGeneratedDocuments, state.GeneratorInfo, finalCompilationWithGeneratedDocuments, intermediateProjects);
+                            withCompilationState.CompilationWithoutGeneratedDocuments,
+                            state.GeneratorInfo,
+                            finalCompilationWithGeneratedDocuments,
+                            intermediateProjects);
                         return newState;
                     }
-                    else
+                    else if (state is NoCompilationState)
```

Comment:
because of the split int eh type hierarchy, we now know that if you tested for WithCompilationState, that the only thing left is a NoCompilationState.